### PR TITLE
Create IsCertificateIssuerLE function to determine if certificate is LE

### PR DESCRIPTION
**Pull request by cblecker**
Create a more robust function, with testing, to determine if a certificate is issued by Let's Encrypt.

### CODE REVIEW

File: pkg/leclient/utils.go [hunk 1]
```diff
@@ -36,3 +37,20 @@ func GetSecret(kubeClient client.Client, secretName, namespace string) (*corev1.
 
 	return s, nil
 }
+
+// IsCertificateIssuerLE takes an issuer name on a certificate and determines if it's a Let's Encrypt CA
+func IsCertificateIssuerLE(issuer pkix.Name) bool {
+	if len(issuer.Organization) > 0 {
+		for _, o := range issuer.Organization {
+			if o == "Let's Encrypt" {
```
Comment (dustman9000) [inline pkg/leclient/utils.go:45]: Can we make these constants?

File: pkg/leclient/utils.go [hunk 2]
```diff
@@ -36,3 +37,20 @@ func GetSecret(kubeClient client.Client, secretName, namespace string) (*corev1.
...
+				return true
+			}
+		}
+	}
+
+	if issuer.CommonName == "Fake LE Intermediate X1" {
```
Comment (dustman9000) [inline pkg/leclient/utils.go:51]: Suggest a constant for this string as well...

**Comment by dustman9000** [comment]
Looks good! Just some minor comments on turning those strings into constants and re-using in unit-tests.

File: pkg/leclient/utils.go (hunk 1 above)
Comment (cblecker) [reply pkg/leclient/utils.go:45]: If we aren't reusing them, what's the benefit of a constant?

Comment (cblecker) [reply pkg/leclient/utils.go:45]: I want the tests to be hard coded, as I don't want them touched if we mess with this function (the actual certs won't ever change). The tests also have more data then we actually check in this function.

So I'm intentionally *not* reusing strings with the tests.. and if not there, then we don't really reuse these strings anywhere (so the benefit of allocating a constant isn't there).

Comment (dustman9000) [reply pkg/leclient/utils.go:45]: Fair enough, thanks.

**Pull request merged**